.github/workflows/ci.yml

```diff
@@ -111,8 +111,16 @@ jobs:
           VERSION: "2.2.11"
 
       # Note: the weeder version will need to be updated when we next update our version
-      # of GHC
-      - run: "stack --no-terminal --jobs=2 build --copy-compiler-tool weeder-2.2.0"
+      # of GHC.
+      #
+      # weeder-2.2.0 has somewhat strange version deps. It doesn't appear
+      # to support the versions of dhall and generic-lens in LTS-18.  However,
+      # forcing it to use the versions of dhall and generic-lens in LTS-18
+      # doesn't cause any problems, so the following commands build weeder
+      # while ignoring version constraints.
+      - run: |
+          echo 'allow-newer: true' >> stack.yaml
+          stack --no-terminal --jobs=2 build --copy-compiler-tool weeder-2.2.0
```

Comment:
This `allow-newer: true` change applies to every `stack` invocation downstream of this step, too. This is probably fine—any issues obscured by ignoring package versions would likely be caught in one of the build jobs—but structuring the workflow like this makes it non-obvious. Please choose one:

* Make the `allow-newer: true` apply only to this invocation of `stack`, probably by creating a copy of `stack.yaml` and either restoring it afterward or using the `--stack-yaml` option on this invocation (although if you know a cleaner way, be my guest!)
* Pull the `allow-newer` modification into its own step, and move the explanatory comment there.

In either case, if the `run` property continues to span multiple lines, please provide a friendly `name` property for the step.

Reply:
I noticed this when wrote the above code, but I thought it wouldn't be likely to negatively affect the following calls to `stack`.

Although I'm not that confident about that, so adding `allow-newer: true` only for installing weeder sounds like the safest thing to do.  Using a separate `stack.yaml` for installing weeder sounds like a good idea, so I made this change in 748aefb3.